AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_entity_utils.py

```diff
@@ -750,6 +751,37 @@ def focus_on_owning_prefab(self) -> None:
         focus_prefab_result = azlmbr.prefab.PrefabFocusPublicRequestBus(bus.Broadcast, "FocusOnOwningPrefab", self.id)
         assert focus_prefab_result.IsSuccess(), f"Prefab operation 'FocusOnOwningPrefab' failed. Error: {focus_prefab_result.GetError()}"
 
+    def has_overrides(self) -> bool:
+        """
+        Validates if a given entity has overrides present. NOTE: This should only be used on an entity within a prefab
+        as this will currently always return as True on a container entity.
+        :return: True if overrides are present, False otherwise
+        """
+        return prefab.PrefabOverridePublicRequestBus(bus.Broadcast, "AreOverridesPresent", self.id)
+
+    def revert_overrides(self) -> bool:
+        """
+        Reverts overrides on a given entity if overrides are present
+        :return: True is overrides were detected and reverted, False otherwise
+        """
+        return prefab.PrefabOverridePublicRequestBus(bus.Broadcast, "RevertOverrides", self.id)
+
+    def validate_expected_components(self, expected_components: list = None,
```

Comment:
The issue with using has_components() is this check is repeated several times throughout the newly updated tests to validate expected results after Undo/Redo as well as after reverting overrides on an an entity. That said, I agree it probably doesn't belong in editor_entity_utils, so moving it out to PrefabTestUtils since it very likely will only apply to tests for prefab overrides.